### ISSUE TITLE
When bumping to v1.6.4 it produces a false positive and the command line output differs from IDE

### ISSUE THREAD

**Issue by binarykitchen** [vitest-dev/eslint-plugin-vitest#854]
Hi ya

When I bumped your plugin from/to:

```git
-    "@vitest/eslint-plugin": "1.5.2",
+    "@vitest/eslint-plugin": "1.6.4",
```

It now produces this new error from the command line

```bash
/home/michael-heuberger/code/binarykitchen/videomail.io/src/server/models/__tests__/videomail.test.ts
  248:18  error  Unnecessary async function wrapper  vitest/no-unneeded-async-expect-function

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Which is this code

```ts
/* Line 248 */    await expect(async () => {
/* Line 249 */      await videomail.load();
/* Line 250 */    }).rejects.toThrow(NotFoundError);
```

There is nothing wrong. `videomail.load()` is async, hence async is needed on line 248. Perhaps a false positive?

Second issue: my IDE (latest VS Code version) doesn't display this error. Why?

Happy to provide more code snippets if you need any. Thanks for any advice.

**Comment by binarykitchen** [comment]
Good idea. It applies these changes:

```ts
-    await expect(async () => {
-      await videomail.load();
-    }).rejects.toThrow(NotFoundError);
+    await expect(videomail.load()).rejects.toThrow(NotFoundError);
```

I think it makes sense, it's more an optimisation/simplification.

My only concern is why my IDE didn't display this error?

**Comment by veritem** [comment]
You might want to take a look into your editor configurations. I don't know if there is anything else we can do about that on a plugin level.
